File: Typescript/Questao3.ts
// # Calcular faturamento
import * as json from "./dados.json";

const calcularFaturamento = () => {
  // Importando e filtrando os valores que sao 0 como feriados e finais de semana.
  const d = json;
  const { data } = d;
  const valores = data.map((item) => item.valor).filter((valor) => valor > 0);

  // Obtendo o menor valor
  const menorValor = Math.min(...valores);

  // Obtendo o maior valor
  const maiorValor = Math.max(...valores);

  // Obtendo a media
  const mediaMensal =
    valores.reduce((soma, valor) => soma + valor, 0) / valores.length;

  //Obtendo os dias que venderam mais que a media
  const diasSuperiores = valores.filter((valor) => valor > mediaMensal).length;

  console.log(`Menor valor de faturamento: ${menorValor}`);
  console.log(`Maior valor de faturamento: ${maiorValor}`);
  console.log(
    `Número de dias no mês em que o valor de faturamento diário foi superior à média mensal: ${diasSuperiores}`
  );
};

calcularFaturamento();

File: Python/Questao3.py
import json

with open('dados.json') as f:
    dados = json.load(f)

# Obtendo os valores de faturamento
faturamento_mensal = [dado['valor'] for dado in dados]

# Obtendo os valores mininos e maximos ecluindo fins de semana e feriado
menor_valor = min(filter(lambda x: x > 0, faturamento_mensal))
maior_valor = max(faturamento_mensal)

# Obtendo a media
media_mensal = sum(faturamento_mensal) / len([dia for dia in faturamento_mensal if dia > 0])

# Contando os dias acima da media
dias_acima_da_media = len([dia for dia in faturamento_mensal if dia > media_mensal])


print(f"Menor valor de faturamento: R${menor_valor:.2f}")
print(f"Maior valor de faturamento: R${maior_valor:.2f}")

print(f"Número de dias com faturamento acima da média: {dias_acima_da_media}")